tests/spread/plugins/maven/legacy-build-and-run/task.yaml

```diff
@@ -0,0 +1,15 @@
+summary: Build and run a basic maven snap with no base
+
+environment:
+  SNAP_DIR: ../snaps/legacy-maven-hello
+
+restore: |
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+
+execute: |
+  cd "$SNAP_DIR"
+  snapcraft
+  sudo snap install maven-hello_*.snap --dangerous
+  [ "$(maven-hello)" = "Hello World!" ]
```

Comment:
We've sort of established two different patterns here: legacy pulling, and legacy running (i.e. entire lifecycle and install/run). Is there something gained by doing more than pulling, here?

Reply:
There is nothing and the original tests actually do not even try and run.

Reply:
I think that is what you adopted versus what I adopted
```
tests/spread/plugins/dotnet/legacy-build-run
tests/spread/plugins/waf/legacy-build-run
tests/spread/plugins/ruby/legacy-pull
tests/spread/plugins/maven/snaps/legacy-maven-hello
tests/spread/plugins/maven/legacy-build-and-run
tests/spread/plugins/go/legacy-build-run
tests/spread/plugins/plainbox/legacy-build-run
tests/spread/plugins/meson/snaps/legacy-meson
tests/spread/plugins/meson/legacy-build-run
tests/spread/plugins/autotools/legacy-pull
tests/spread/plugins/catkin/legacy-pull
tests/spread/plugins/ament/legacy-pull
tests/spread/plugins/make/legacy-pull
tests/spread/plugins/scons/legacy-build-run
tests/spread/plugins/cmake/legacy-pull
tests/spread/plugins/godeps/legacy-build-run
```
With the exception of ament where I admit I just did pull to save time.